drivers/entropy/CMakeLists.txt

```diff
@@ -1,10 +1,20 @@
+if (CONFIG_ENTROPY_ESP32_RNG OR
```

Comment:
This `if ... endif` looks unnecessary. The `zephyr_library_sources_ifdef()` does the same thing.

Reply:
The thing is that zephyr_library() function forces to have at least one source file for the library.
Removing zephyr_library() will cause the following error:

> CMake Error at ../../../cmake/extensions.cmake:360 (add_library):
>   Cannot find source file:
> 
>     entropy_nrf5.c

This is the bad hack to avoid the problem, but still have ENTROPY_HAS_DRIVER enabled. I've asked about how to solve this.